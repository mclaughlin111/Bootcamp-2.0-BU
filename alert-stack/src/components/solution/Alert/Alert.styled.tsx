import styled from "styled-components";

export const StyledAlert = styled.div`
  color: black;
  border-radius: 8px;

  padding: 20px 10px 20px 20px;
  margin-bottom: 10px;
  display: flex;
  background-color: lightgray;
`;

export const StyledCloseButton = styled.svg`
  width: 24px;
  padding-left: 20px;
  cursor: pointer;
`;
