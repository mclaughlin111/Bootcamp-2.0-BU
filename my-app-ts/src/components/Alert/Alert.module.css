.alert {
  color: black;
  border-radius: 8px;
  padding: 20px;
}

.info {
  background-color: lightblue;
}

.error {
  background-color: red;
}

.warning {
  background-color: orange;
}

.success {
  background-color: lightgreen;
}
