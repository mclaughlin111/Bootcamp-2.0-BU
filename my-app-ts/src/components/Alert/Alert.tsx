import styles from "./Alert.module.css";
import classNames from "classnames";

export type AlertType = "error" | "warning" | "info" | "success";

type AlertProps = {
  message: string;
  type?: AlertType;
};

export const Alert = ({ message, type = "info" }: AlertProps) => (
  <div className={classNames(styles.alert, styles[type])}>{message}</div>
);
