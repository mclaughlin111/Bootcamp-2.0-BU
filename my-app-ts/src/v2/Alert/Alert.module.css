.alert {
  color: black;
  border-radius: 8px;
  padding: 20px 10px 20px 20px;
  margin-bottom: 10px;
  display: flex;
}

.alert svg {
  width: 24px;
  padding-left: 20px;
}

.close {
  cursor: pointer;
}

.info {
  background-color: lightblue;
}

.error {
  background-color: red;
}

.warning {
  background-color: orange;
}

.success {
  background-color: lightgreen;
}
