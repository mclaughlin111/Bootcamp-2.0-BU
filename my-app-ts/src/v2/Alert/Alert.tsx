/* eslint-disable jsx-a11y/anchor-is-valid */
import styles from "./Alert.module.css";
import classNames from "classnames";

export type AlertType = "error" | "warning" | "info" | "success";

type AlertProps = {
  message: string;
  type?: AlertType;
};

export const Alert = ({ message, type = "info" }: AlertProps) => {
  return (
    <div className={classNames(styles.alert, styles[type])}>
      <div>{message}</div>
      <div
        className={styles.close}
        onClick={() => console.log("clicked Close")}
      >
        <svg
          fill="black"
          xmlns="http://www.w3.org/2000/svg"
          viewBox="0 0 1792 1792"
        >
          <path
            d="M1082.2,896.6l410.2-410c51.5-51.5,51.5-134.6,0-186.1s-134.6-51.5-186.1,0l-410.2,410L486,300.4
   c-51.5-51.5-134.6-51.5-186.1,0s-51.5,134.6,0,186.1l410.2,410l-410.2,410c-51.5,51.5-51.5,134.6,0,186.1
   c51.6,51.5,135,51.5,186.1,0l410.2-410l410.2,410c51.5,51.5,134.6,51.5,186.1,0c51.1-51.5,51.1-134.6-0.5-186.2L1082.2,896.6z"
          />
        </svg>
      </div>
    </div>
  );
};
