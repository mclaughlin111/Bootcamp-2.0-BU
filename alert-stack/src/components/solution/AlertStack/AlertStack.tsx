import { useState } from "react";
import { Alert, MessageAlert } from "../Alert/Alert";

type AlertStackProps = {
  alerts: MessageAlert[];
};

export const AlertStack = ({ alerts }: AlertStackProps) => {
  const [myAlerts, setMyAlerts] = useState(alerts);

  return (
    <>
      {myAlerts.map((alert, index) => {
        return (
          <Alert
            key={index}
            id={index}
            message={alert.message}
            type={alert.type}
            onClosed={(id) =>
              setMyAlerts((alerts) => alerts.filter((_, index) => index !== id))
            }
          />
        );
      })}
    </>
  );
};
