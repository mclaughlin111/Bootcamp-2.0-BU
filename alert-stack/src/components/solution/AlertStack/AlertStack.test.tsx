import { render, screen, within } from "@testing-library/react";
import { MessageAlert } from "../Alert/Alert";
import { AlertStack } from "./AlertStack";
import userEvent from "@testing-library/user-event";

const testAlerts: MessageAlert[] = [
  { message: "Test info alert" },
  { message: "test error alert", type: "error" },
  { message: "test error alert warning", type: "warning" },
  { message: "test error alert success", type: "success" },
];

test("alert stack with array of alerts", () => {
  const { container } = render(<AlertStack alerts={testAlerts} />);

  expect(container).toMatchSnapshot();
});

test("Closing an alert", async () => {
  const warningAlertIndex = 2;

  render(<AlertStack alerts={testAlerts} />);

  const alerts = screen.getAllByRole("alert");

  const alertButton = within(alerts[warningAlertIndex]).getByRole("button");
  await userEvent.click(alertButton);

  const afterClickAlerts = screen.getAllByRole("alert");
  expect(afterClickAlerts.length).toEqual(3);
  expect(screen.queryByText("test error alert warning")).toBeNull();
});
