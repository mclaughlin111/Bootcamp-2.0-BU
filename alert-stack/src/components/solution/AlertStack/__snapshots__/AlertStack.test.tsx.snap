// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`alert stack with array of alerts 1`] = `
<div>
  <div
    class="sc-aYaIB XnGOZ"
    role="alert"
  >
    <div>
      Test info alert
    </div>
    <div
      role="button"
    >
      <svg
        class="sc-gEvDqW bLKDy"
        fill="black"
        viewBox="0 0 1792 1792"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1082.2,896.6l410.2-410c51.5-51.5,51.5-134.6,0-186.1s-134.6-51.5-186.1,0l-410.2,410L486,300.4 c-51.5-51.5-134.6-51.5-186.1,0s-51.5,134.6,0,186.1l410.2,410l-410.2,410c-51.5,51.5-51.5,134.6,0,186.1 c51.6,51.5,135,51.5,186.1,0l410.2-410l410.2,410c51.5,51.5,134.6,51.5,186.1,0c51.1-51.5,51.1-134.6-0.5-186.2L1082.2,896.6z"
        />
      </svg>
    </div>
  </div>
  <div
    class="sc-aYaIB XnGOZ"
    role="alert"
  >
    <div>
      test error alert
    </div>
    <div
      role="button"
    >
      <svg
        class="sc-gEvDqW bLKDy"
        fill="black"
        viewBox="0 0 1792 1792"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1082.2,896.6l410.2-410c51.5-51.5,51.5-134.6,0-186.1s-134.6-51.5-186.1,0l-410.2,410L486,300.4 c-51.5-51.5-134.6-51.5-186.1,0s-51.5,134.6,0,186.1l410.2,410l-410.2,410c-51.5,51.5-51.5,134.6,0,186.1 c51.6,51.5,135,51.5,186.1,0l410.2-410l410.2,410c51.5,51.5,134.6,51.5,186.1,0c51.1-51.5,51.1-134.6-0.5-186.2L1082.2,896.6z"
        />
      </svg>
    </div>
  </div>
  <div
    class="sc-aYaIB XnGOZ"
    role="alert"
  >
    <div>
      test error alert warning
    </div>
    <div
      role="button"
    >
      <svg
        class="sc-gEvDqW bLKDy"
        fill="black"
        viewBox="0 0 1792 1792"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1082.2,896.6l410.2-410c51.5-51.5,51.5-134.6,0-186.1s-134.6-51.5-186.1,0l-410.2,410L486,300.4 c-51.5-51.5-134.6-51.5-186.1,0s-51.5,134.6,0,186.1l410.2,410l-410.2,410c-51.5,51.5-51.5,134.6,0,186.1 c51.6,51.5,135,51.5,186.1,0l410.2-410l410.2,410c51.5,51.5,134.6,51.5,186.1,0c51.1-51.5,51.1-134.6-0.5-186.2L1082.2,896.6z"
        />
      </svg>
    </div>
  </div>
  <div
    class="sc-aYaIB XnGOZ"
    role="alert"
  >
    <div>
      test error alert success
    </div>
    <div
      role="button"
    >
      <svg
        class="sc-gEvDqW bLKDy"
        fill="black"
        viewBox="0 0 1792 1792"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1082.2,896.6l410.2-410c51.5-51.5,51.5-134.6,0-186.1s-134.6-51.5-186.1,0l-410.2,410L486,300.4 c-51.5-51.5-134.6-51.5-186.1,0s-51.5,134.6,0,186.1l410.2,410l-410.2,410c-51.5,51.5-51.5,134.6,0,186.1 c51.6,51.5,135,51.5,186.1,0l410.2-410l410.2,410c51.5,51.5,134.6,51.5,186.1,0c51.1-51.5,51.1-134.6-0.5-186.2L1082.2,896.6z"
        />
      </svg>
    </div>
  </div>
</div>
`;
